agent/worker.go

```diff
@@ -568,9 +573,9 @@ func (w *worker) Subscribe(ctx context.Context, subscription *api.SubscriptionMe
 		case v := <-ch:
 			task := v.(*api.Task)
 			if match(task) {
-				w.mu.Lock()
+				w.mu.RLock()
 				go w.taskManagers[task.ID].Logs(ctx, *subscription.Options, publisher)
```

Comment:
There might be another bug here that we may want to fix: what if there is already an existing logs call for a given service? This is probably okay, in practice. Leave that for another PR.